src/editors/sharedComponents/TinyMceWidget/hooks.js

```diff
@@ -151,6 +156,19 @@ export const setupCustomBehavior = ({
       editor, images, setImage, openImgModal,
     }),
   });
+
+  // insert link button
+  editor.ui.registry.addButton(tinyMCE.buttons.insertLink, {
+    icon: 'new-tab',
+    tooltip: translations?.insertLinkTooltipTitle ?? '',
+    onAction: openInsertLinkModal,
+    onSetup(api) {
+      editor.on('SelectionChange', () => {
+        api.setDisabled(editor.selection.getContent().length === 0);
```

Comment:
```suggestion
        api.setDisabled(
          editor.selection.getContent().length === 0 &&
          editor.dom.getParent(editor.selection.getNode(), 'a[href]') === null
        );
```

This suggestion does not include logic to check if the link is an internal "jump to" link or an external link. I think it should be possible to create a helper function with that logic and utilize it both here and when getting the block id the `useEffect()` in `InsertLinkModal/index.jsx`

Reply:
This line is not working for me, but I found the part to check if a link has already added
`editor.dom.getParent(editor.selection.getNode(), 'a[href]') === null`